components/style/properties/shorthand/box.mako.rs

```diff
@@ -308,28 +308,88 @@ macro_rules! try_parse_one {
 
     impl<'a> LonghandsToSerialize<'a>  {
         fn to_css_declared<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-            try!(self.animation_duration.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_timing_function.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_delay.to_css(dest));
-            try!(write!(dest, " "));
-
-            try!(self.animation_direction.to_css(dest));
-            try!(write!(dest, " "));
+            fn extract_value<T>(x: &DeclaredValue<T>) -> Option< &T> {
+                match *x {
+                    DeclaredValue::Value(ref val) => Some(val),
+                    _ => None,
+                }
+            }
 
-            try!(self.animation_fill_mode.to_css(dest));
-            try!(write!(dest, " "));
+            let len = extract_value(self.animation_name).map(|i| i.0.len()).unwrap_or(0);
+            // There should be at least one declared value
+            if len == 0 {
+                return dest.write_str("")
+            }
 
-            try!(self.animation_iteration_count.to_css(dest));
-            try!(write!(dest, " "));
+            // usuper says on #15398: For computation, the value should be cycled, but for
```

Comment:
Nit: Please change this comment to something like,
`// serialize only when all the lists have same length`